render.js

```diff
@@ -0,0 +1,19 @@
+const Renderer = () => {
+    const renderPosts = (posts) => {
```

Comment:
In general, you should add at least 2 more helper functions to separate some of the different concerns here.